ha/k8s_setup/ha_setup.py

```diff
@@ -370,15 +372,26 @@ def _add_cluster_component_health(self) -> None:
                                    specific_info=specific_info)
 
 
-    def _add_node_health(self, data_node_ids, server_node_ids, control_node_ids) -> None:
+    def _add_node_health(self, data_node_ids, server_node_ids, control_node_ids, kv_enable_batch_put: bool = True) -> None:
         """
         Add node health
         """
         _, _, node_mapping = self._confStoreAPI.get_cluster_cardinality()
+        node_id_list = []
+        node_id_len = len(node_mapping)
         # cluster cardinality node_to_name mapping will have actual machine-ids
         # If that is part of one of the list from data, server or control node
         # list, then mark the respective functional type and update the specific info
         for node_id in node_mapping.values():
+            node_id_list.append(node_id)
+            if len(node_id_list) > CONSUL_TRANSACTIONS_LIMIT:
+                # Note: if batch put is enabled, need to commit
+                # to push all the local cashed values to consul server
+                if kv_enable_batch_put:
+                    Log.debug(f"pushing {CONSUL_TRANSACTIONS_LIMIT} node health transactions to store")
+                    self._confstore.commit()
+                node_id_list.clear()
+                node_id_list.append(node_id)
```

Comment:
Am I missing something here - why are we appending node_id 'again' to node_id_list?

Reply:
node_id is coming from the node_mapping dictionary. I have explicitly introduced the node_id list here for allowing 64 transactions in one consul transaction request.
If the length of this list exceeds (>64), This confstore.commit() will be called which will push 64 keys to kv store. (So, for every 64 bunch, this will happen).

Reply:
OK, make sense.

Reply:
Sure, Thanks!